document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java

```diff
@@ -409,10 +408,13 @@ public boolean deleteAll() {
     return true;
   }
 
-  private void parseOrderByQuery(List<OrderBy> orderBys, Map<String, Object> orderbyMap) {
+  @VisibleForTesting
+  LinkedHashMap<String, Object> parseOrderByQuery(List<OrderBy> orderBys) {
```

Comment:
nit: perhaps a query parser util/class would better follow SRP and remove the need for `@VisibleForTesting`, everyone's favorite antipattern 😉

Reply:
Moved the query parsing methods into QueryParser classes - one for mongo and another for postgres.